.github/workflows/theme.yml

```diff
@@ -0,0 +1,83 @@
+name: Theme
+on:
+  workflow_call:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    name: Build Theme
+    runs-on: ${{ github.event.act && 'ubuntu-latest' || format('runs-on={0}/runner=sw-amd64', github.run_id) }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Build Theme
+        uses: shopware/github-actions/build-zip@main
+        env:
+          NODE_OPTIONS: "--openssl-legacy-provider"
+        with:
+          extensionName: '${{ github.event.repository.name }}'
+
+  install:
+    name: Test Theme installation
+    runs-on: ${{ github.event.act && 'ubuntu-latest' || format('runs-on={0}/runner=sw-amd64', github.run_id) }}
+    needs:
+      - build
+    services:
+      mysql:
+        image: mysql:8.0
+        env:
+          MYSQL_ALLOW_EMPTY_PASSWORD: 'yes'
+          MYSQL_DATABASE: shopware
+        ports:
+          - 3306:3306
+        options: --health-cmd="mysqladmin ping" --health-interval=5s --health-timeout=2s --health-retries=3
+        volumes:
+          - /dev/shm/mysql:/var/lib/mysql
+    steps:
+      - name: Setup and Install Theme
+        uses: shopware/github-actions/setup-extension@main
+        with:
+          mysqlVersion: 'skip'
+          install: 'true'
+          env: 'prod'
+          extensionName: '${{ github.event.repository.name }}'
+          extension-zip: '${{ github.event.repository.name }}.zip'
+
+      - name: Validate Snippets
+        run: |-
+          bin/console snippets:validate
+
+  should_test_jest:
+    name: Determine whether to run Jest tests
+    runs-on: ${{ github.event.act && 'ubuntu-latest' || format('runs-on={0}/runner=sw-amd64', github.run_id) }}
+    outputs:
+      test_jest: ${{ steps.should_run.outputs.test_jest }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Determine whether to run Jest tests
+        id: should_run
+        run: |-
+          echo "test_jest=${{ hashFiles('Resources/app/storefront/package-lock.json') != '' && hashFiles('Resources/app/storefront/jest.config.*') != '' }}" >> ${GITHUB_OUTPUT}
```

Comment:
I feel like for a single job there should be simpler solution.

Reply:
Yes, I think so, too. This workflow should be reusable, though, so we need to determine whether there are tests to be executed.

The `hashFiles` function is [unfortunately only available](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#context-availability) below the `jobs.<job_id>.steps` context and it is the only way to check for file availability directly within GHA.

So the alternative would be to add an `if: hashFiles(...)` to _every step_ of the test job, hence why I went with this approach.